harness/determined/profiler.py

```diff
@@ -0,0 +1,524 @@
+import threading
+import psutil
+import pynvml
+import datetime
+import queue
+import time
+from determined.common import api
+
+
+SYSTEM_METRIC_TYPE_ENUM = "PROFILER_METRIC_TYPE_SYSTEM"
+
+
+
+
+class ProfilerAgent:
+    """
+    Agent that collects metrics and sends them to the master. It has:
```

Comment:
can we drop a comment about all the threads there are and which starts/joins which just for readers?